src/ui/src/builder/useComponentFieldViewModel.ts

```diff
@@ -0,0 +1,45 @@
+import { computed, inject, Ref, DeepReadonly } from "vue";
+import injectionKeys from "@/injectionKeys";
+import { useComponentActions } from "./useComponentActions";
+
+type Params = {
+	componentId: DeepReadonly<Ref<string>>;
+	fieldKey: DeepReadonly<Ref<string>>;
+	defaultValue?: DeepReadonly<Ref<string>>;
+};
```

Comment:
I think you want to use `MaybeRef` for `fieldKey`
```suggestion
type Params = {
	componentId: DeepReadonly<Ref<string>>;
	fieldKey: MaybeRef<string>;
	defaultValue?: DeepReadonly<Ref<string>>;
};
```

It makes your function accept `string | Ref<string>`, so you don't have to wrap your string like this

https://github.com/writer/writer-framework/blob/f9470c683698a0b2f3bb8f181179a9e432d3252f/src/ui/src/builder/settings/BuilderFieldsWriterResourceId.vue#L99

Then you just need to use `unref()` when you want to use the value

Reply:
great suggestion!
I haven't thought about that.